src/auth/strategies/jwt.strategy.ts

```diff
@@ -15,7 +15,10 @@ export class UserAuthBearerStrategy extends PassportStrategy(
       secretOrKey: jwtConstants.secret,
     });
   }
-  validate(payload: JwtPayload): JwtPayload {
+  validate(payload: any): JwtPayload {
+    if (payload.status === 'FURTHER_ACTION_IS_REQUIRED') {
+      throw new ForbiddenException(payload.message);
+    }
```

Comment:
hmm I don't understand this method - it says `validate` but we are checking the payload. We should not rely on payload for validation but on the JWT signature that every JWT token should have.

Also `JwtPayload` changed to `any` - I have a hunch that you are using this method for both bearer jwt token and 2fa bearer token. Don't do it like this. We should have a separate validate method for each JWT type. For each we should try and validate them using their signatures. If the signature is invalid then obviously it's not a valid token of this type.